src/domain/usecases/data-entry/amc/ImportAMCProductLevelData.ts

```diff
@@ -151,48 +193,127 @@ export class ImportAMCProductLevelData {
+                            status: "COMPLETED",
                         };
                     });
 
                     const enrollments: D2TrackerEnrollment[] = [
                         {
-                            orgUnit: orgUnitId,
+                            orgUnit: tei.orgUnit.id,
                             program: AMC_PRODUCT_REGISTER_PROGRAM_ID,
                             enrollment: "",
                             trackedEntityType: AMR_GLASS_AMC_TET_PRODUCT_REGISTER,
                             notes: [],
                             relationships: [],
                             attributes: attributes,
                             events: events,
-                            enrolledAt: new Date().getTime().toString(),
-                            occurredAt: new Date().getTime().toString(),
+                            enrolledAt: tei.enrollment?.enrollmentDate ?? new Date().getTime().toString(),
+                            occurredAt: tei.enrollment?.incidentDate ?? new Date().getTime().toString(),
                             createdAt: new Date().getTime().toString(),
                             createdAtClient: new Date().getTime().toString(),
                             updatedAt: new Date().getTime().toString(),
                             updatedAtClient: new Date().getTime().toString(),
                             status: "ACTIVE",
-                            orgUnitName: orgUnitName,
+                            orgUnitName: "",
                             followUp: false,
                             deleted: false,
                             storedBy: "",
                         },
                     ];
                     const entity: D2TrackerTrackedEntity = {
-                        orgUnit: orgUnitId,
+                        orgUnit: tei.orgUnit.id,
                         trackedEntity: "",
                         trackedEntityType: AMR_GLASS_AMC_TET_PRODUCT_REGISTER,
                         enrollments: enrollments,
-                        attributes: [],
+                        attributes: attributes.map(attr => {
+                            return {
+                                attribute: attr.attribute,
+                                value: attr.value.toString(),
+                            };
+                        }),
                     };
+
                     return entity;
                 });
                 return Future.success(trackedEntities);
             });
     }
+
+    private validateTEIsAndEvents(
+        teis: D2TrackerTrackedEntity[],
+        orgUnitId: string,
+        orgUnitName: string,
+        period: string,
+        programId: string
+    ): FutureData<ValidationResult> {
+        //1. Before running validations, add ids to tei, enrollement and event so thier relationships can be processed.
+        const teisWithId = teis?.map((tei, teiIndex) => {
```

Comment:
Here in all this map() functions is better to use destructuring to create new variable `teisWithId` instead of updating in place to avoid changing `teis` variable

Reply:
code updated to address this